internal/services/datadog/datadog_monitors_resource_test.go

```diff
@@ -0,0 +1,294 @@
+			"datadog_organization_properties.0.api_key",
+			"datadog_organization_properties.0.application_key",
+			"datadog_organization_properties.0.enterprise_app_id",
+			"datadog_organization_properties.0.linking_auth_code",
+			"datadog_organization_properties.0.linking_client_id",
+			"datadog_organization_properties.0.redirect_uri"),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("user_info",
+			"user_info.0.name",
+			"user_info.0.email_address",
+			"datadog_organization_properties",
+			"datadog_organization_properties.0",
+			"datadog_organization_properties.0.id",
+			"datadog_organization_properties.0.name",
+			"datadog_organization_properties.0.api_key",
+			"datadog_organization_properties.0.application_key",
+			"datadog_organization_properties.0.enterprise_app_id",
+			"datadog_organization_properties.0.linking_auth_code",
+			"datadog_organization_properties.0.linking_client_id",
+			"datadog_organization_properties.0.redirect_uri"),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep("user_info",
+			"user_info.0.name",
+			"user_info.0.email_address",
+			"datadog_organization_properties",
+			"datadog_organization_properties.0",
+			"datadog_organization_properties.0.id",
+			"datadog_organization_properties.0.name",
+			"datadog_organization_properties.0.api_key",
+			"datadog_organization_properties.0.application_key",
+			"datadog_organization_properties.0.enterprise_app_id",
+			"datadog_organization_properties.0.linking_auth_code",
+			"datadog_organization_properties.0.linking_client_id",
+			"datadog_organization_properties.0.redirect_uri"),
+	})
+}
+
+func (r DatadogMonitorResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.DatadogMonitorID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Datadog.MonitorsClient.Get(ctx, id.ResourceGroup, id.MonitorName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving Datadog Monitor %q (Resource Group %q): %+v", id.MonitorName, id.ResourceGroup, err)
+	}
+	return utils.Bool(true), nil
+}
+
+func (r DatadogMonitorResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-datadog-%d"
+  location = "%s"
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r DatadogMonitorResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+	%s
+	
+	resource "azurerm_datadog_monitor" "test" {
+		name = "test-terraform-%d"
+		resource_group_name = azurerm_resource_group.test.name
+		location = "EAST US 2 EUAP"
```

Comment:
We should not hardcode a location here if it's global released, how about `location = azurerm_resource_group.test.location`?

Reply:
We are only available in West US 2 region

Reply:
Got it, thanks!